RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h

```diff
@@ -0,0 +1,453 @@
+{
+public:
+
+  enum class OperationMode {BEAM_SPOT_FIXED, BEAM_SPOT_SIGMA, VERTICES_FIXED, VERTICES_SIGMA };
+  enum class SeedingMode {CANDIDATE_SEEDED, POINT_SEEDED, CANDIDATE_POINT_SEEDED};
+
+  explicit CandidatePointSeededTrackingRegionsProducer(const edm::ParameterSet& conf,
+	edm::ConsumesCollector && iC)
+  {
+    edm::ParameterSet regPSet = conf.getParameter<edm::ParameterSet>("RegionPSet");
+
+    // operation mode
+    std::string operationModeString       = regPSet.getParameter<std::string>("operationMode");
+    if      (operationModeString == "BeamSpotFixed") m_operationMode = OperationMode::BEAM_SPOT_FIXED;
+    else if (operationModeString == "BeamSpotSigma") m_operationMode = OperationMode::BEAM_SPOT_SIGMA;
+    else if (operationModeString == "VerticesFixed") m_operationMode = OperationMode::VERTICES_FIXED;
+    else if (operationModeString == "VerticesSigma") m_operationMode = OperationMode::VERTICES_SIGMA;
+    else throw edm::Exception(edm::errors::Configuration) << "Unknown operation mode string: "<<operationModeString;
+
+    // seeding mode
+    std::string seedingModeString = regPSet.getParameter<std::string>("seedingMode");
+    if      (seedingModeString == "Candidate")      m_seedingMode = SeedingMode::CANDIDATE_SEEDED;
+    else if (seedingModeString == "Point")          m_seedingMode = SeedingMode::POINT_SEEDED;
+    else if (seedingModeString == "CandidatePoint") m_seedingMode = SeedingMode::CANDIDATE_POINT_SEEDED;
+    else throw edm::Exception(edm::errors::Configuration) << "Unknown seeding mode string: "<<seedingModeString;
+
+    // basic inputs
+    if(m_seedingMode == SeedingMode::CANDIDATE_SEEDED || m_seedingMode == SeedingMode::CANDIDATE_POINT_SEEDED)
+      token_input        = iC.consumes<reco::CandidateView>(regPSet.getParameter<edm::InputTag>("input"));
+
+    // Specific points in the detector
+    if(m_seedingMode == SeedingMode::POINT_SEEDED || m_seedingMode == SeedingMode::CANDIDATE_POINT_SEEDED){
+      edm::ParameterSet points = regPSet.getParameter<edm::ParameterSet>("points");
+      etaPoints = points.getParameter<std::vector<double>>("eta");
+      phiPoints = points.getParameter<std::vector<double>>("phi");
+      n_points = etaPoints.size();
+
+      if (!(etaPoints.size() == phiPoints.size()))  throw edm::Exception(edm::errors::Configuration) << "The parameters 'eta' and 'phi' must have the same size";
+      if (n_points == 0) throw edm::Exception(edm::errors::Configuration) << "At least one point should be defined for point or candidate+point seeding modes";
+
+      for(size_t i = 0; i < n_points; ++i ){
+
+      	double x = std::cos(phiPoints[i]);
+	double y = std::sin(phiPoints[i]);
+	double theta = 2*std::atan(std::exp(-etaPoints[i]));
+	double z = 1./std::tan(theta);
+	GlobalVector direction( x,y,z );
+	directionPoints.push_back(direction);
+
+      }
+
+    }
+
+    m_maxNRegions      = regPSet.getParameter<int>("maxNRegions");
```

Comment:
Given the implementation below, I imagine it can be problematic if m_maxNRegions is below or equal 0.
It would be probably worth putting a check already here in the constructor and throw if m_maxNRegions<=0
Further, I would define as "unsigned int" both m_maxNRegions and the local variable n_regions below

Reply:
Why not changing the configuration parameter type to `unsigned int` then? (the configration compatibility is already broken wrt. `CandidateSeeded...` and `PointSeeded...` with the `deltaEta` and `deltaPhi`)

Reply:
Yes, that was also already in my suggestion, indeed
Still, a check for it being different from 0 is also needed, because the checks in the loops below are only done after having push_back